SymbolLayer.tsx causes linting errors

## Environment
- Dev OS: OSX 12.5.1
- react-native: 0.68.2
- react: 17.0.2

When I run `yarn tsc` while using `SymbolLayer` from `'@rnmapbox/maps'`, I get a typescript error because the code on [line 90](https://github.com/rnmapbox/maps/blob/main/javascript/components/SymbolLayer.tsx#L90) assumes that `type` is defined on `child`, which it might not be.

The error looks like this:
```
node_modules/@rnmapbox/maps/javascript/components/SymbolLayer.tsx:90:18 - error TS2339: Property 'type' does not exist on type 'Element | (Element & string) | (Element & number) | (Element & false) | (Element & true) | (Element & ReactElement<any, string | JSXElementConstructor<...>>) | ... 8 more ... | (Element[] & ReactPortal)'.
  Property 'type' does not exist on type 'Element[] & string'.

90       if (child?.type === View) {
                    ~~~~


Found 1 error in node_modules/@rnmapbox/maps/javascript/components/SymbolLayer.tsx:90

error Command failed with exit code 2.
```

## Steps to reproduce
`yarn tsc`

Minimal Example:

App.tsx:
```
import React from 'react';
import {View} from 'react-native';

import MapboxGL from '@rnmapbox/maps';

const App = () => {
  return (
    <View>
      <MapboxGL.MapView>
        <MapboxGL.ShapeSource id="sample-source">
          <MapboxGL.SymbolLayer
            id="all-visible-blocks-symbols"
            style={{
              textField: 'Hello',
            }}
          />
        </MapboxGL.ShapeSource>
      </MapboxGL.MapView>
    </View>
  );
};

export default App;
```

tsconfig.json:
```
{
  "compilerOptions": {
    /* Basic Options */
    "target": "esnext" /* Specify ECMAScript target version: 'ES3' (default), 'ES5', 'ES2015', 'ES2016', 'ES2017','ES2018' or 'ESNEXT'. */,
    "module": "commonjs" /* Specify module code generation: 'none', 'commonjs', 'amd', 'system', 'umd', 'es2015', or 'ESNext'. */,
    "lib": [
      "es2017"
    ] /* Specify library files to be included in the compilation. */,
    "allowJs": true /* Allow javascript files to be compiled. */,
    // "checkJs": true,                       /* Report errors in .js files. */
    "jsx": "react-native" /* Specify JSX code generation: 'preserve', 'react-native', or 'react'. */,
    // "removeComments": true,                /* Do not emit comments to output. */
    "noEmit": true /* Do not emit outputs. */,
    "isolatedModules": true /* Transpile each file as a separate module (similar to 'ts.transpileModule'). */,

    /* Strict Type-Checking Options */
    "strict": true /* Enable all strict type-checking options. */,

    /* Module Resolution Options */
    "moduleResolution": "node" /* Specify module resolution strategy: 'node' (Node.js) or 'classic' (TypeScript pre-1.6). */,
    "baseUrl": "." /* Base directory to resolve non-absolute module names. */,
    "paths": {
      "~*": ["./src/*"]
    } /* A series of entries which re-map imports to lookup locations relative to the 'baseUrl'. */,
    "allowSyntheticDefaultImports": true /* Allow default imports from modules with no default export. This does not affect code emit, just typechecking. */,
    "esModuleInterop": true /* Enables emit interoperability between CommonJS and ES Modules via creation of namespace objects for all imports. Implies 'allowSyntheticDefaultImports'. */,
    // "preserveSymlinks": true,              /* Do not resolve the real path of symlinks. */
    "skipLibCheck": true,                    /* Skip type checking of declaration files. */
  },
  "exclude": [
    "node_modules",
  ]
}
```

--- a/javascript/components/SymbolLayer.tsx
+++ b/javascript/components/SymbolLayer.tsx
@@ -82,12 +82,12 @@ export class SymbolLayer extends AbstractLayer<Props, NativeTypeProps> {
   _shouldSnapshot() {
     let isSnapshot = false;
 
-    if (React.Children.count(this.props.children) <= 0) {
+    if (React.Children.count(this.baseProps.children) <= 0) {
       return isSnapshot;
     }
 
-    React.Children.forEach(this.props.children, (child) => {
-      if (child !== undefined && 'type' in child && child.type === View) {
+    React.Children.forEach(this.baseProps.children, (child) => {
+      if (child?.type === View) {
         isSnapshot = true;
       }
     });

